src/appliance-onboarding-script/appliance_setup/avs/avsarconboarder/utils/user_config_validatator.py

```diff
@@ -1,25 +1,37 @@
+from operator import truediv
 from avs.avsarconboarder.entity.CustomerResource import CustomerResource
+from avs.avsarconboarder.orchestrator.network._network_orchestrator import NetworkOrchestrator
 from avs.avsarconboarder.exception import InvalidInputError
 from avs.avsarconboarder.retriever.cloud_data.cloud_data_retriever import CloudDataRetriever
-
+from avs.avsarconboarder.constants import Constant
+from ..processor.nsx.helper import _NSXSegmentHelper
 
 class ConfigValidator:
     def __init__(self, config):
         self.__config = config
+        self._network_orchestrator = NetworkOrchestrator()
+        self._segment_helper = _NSXSegmentHelper.NSXSegmentHelper()
+        
+    def validate_static_ip_nw_config_v1(self):
+        self._network_orchestrator.validate_ip_address(self.__config["applianceControlPlaneIpAddress"])
+        self._network_orchestrator.validate_static_ip_cidr_block(self.__config["staticIpNetworkDetails"]["networkCIDRForApplianceVM"], Constant.CONFIG_VERSION_V1)
+        self._network_orchestrator.validate_ip_address(self.__config["staticIpNetworkDetails"]["k8sNodeIPPoolStart"])
+        self._network_orchestrator.validate_ip_address(self.__config["staticIpNetworkDetails"]["k8sNodeIPPoolEnd"])
+        self._network_orchestrator.validate_ip_address(self.__config["staticIpNetworkDetails"]["gatewayIPAddress"])
+        
+        gateway_ip_cidr = self._network_orchestrator.get_gateway_address_cidr_from_network_addr(self.__config["staticIpNetworkDetails"]["networkCIDRForApplianceVM"])
+        if gateway_ip_cidr[0] != self.__config["staticIpNetworkDetails"]["gatewayIPAddress"]:
```

Comment:
>if gateway_ip_cidr[0]  [](http://example.com/codeflow?start=8&length=22)

Can be taken as a separate change - 

Should we also validate the size of the cidr? I faintly remember discussing there could be multiple such scenarios to validate for the older config and we concluded to take them separately. Do we have those noted somewhere or do we have a task to track those?

Reply:
I can create a task. We thought of if the customer gives input as '171.1.1.1/28/30' or '171.1.1.1/33' or '171.1.1./28'. I have added a validation. It should cover all the scenarios.

Reply:
I have validated using ipaddress.ip_network()